checker.go

```diff
@@ -6,6 +6,14 @@ import (
 	"strings"
 )
 
+type sign int
```

Comment:
I really want to merge this code. And you were thoughtful of making these features private. So I will merge as it is, and change the implementation to get rid of this FlagArgument later.